src/connectors/utils.js

```diff
@@ -161,10 +171,45 @@ export async function queryRequest(endpoint, endpointHost, qps = {}) {
     pms.set('limit', '500');
   }
   const flag = `${endpoint}Flag`;
-  const checkData = () => {
-    if (Object.hasOwn(window, flag) && window[flag] === true) {
+  const checkData = async () => {
+     if(type === 'submit'){
+      await fetch(`${endpointHost}${endpoint}?${pms.toString()}`)
+          .then((resp) => resp.json())
+          .then((data) => {
+            window[flag] = false;
+            if (!Object.hasOwn(window, 'dashboard')) {
+              window.dashboard = {};
+            }
+            window.dashboard[endpoint+"-"+type] = data;
+          })
+          .catch((err) => {
+            // eslint-disable-next-line no-console
+            console.error('API Call Has Failed, Check that inputs are correct', err.message);
+          });
+    } 
+    else if(type === 'cwv'){
+      endpoint ="rum-dashboard";
+      await fetch(`${endpointHost}${endpoint}?${pms.toString()}`)
+          .then((resp) => resp.json())
+          .then((data) => {
+            console.log("indie cwv data");
+            console.log(data);
+            window[flag] = false;
+            if (!Object.hasOwn(window, 'dashboard')) {
+              window.dashboard = {};
```

Comment:
I'd recommend using the state manager instead of storing these in window global object.